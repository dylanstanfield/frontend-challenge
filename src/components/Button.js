import { styled, Button as MuiButton } from '@mui/material';

const CustomButton = styled(MuiButton)(({ theme }) => `
  text-transform: none;
  box-shadow: none;
  font-weight: 800;

  &:hover {
    box-shadow: none;
  }
`);

export const Button = ({ children, color, onClick, disabled, testId }) => (
  <CustomButton
    variant="contained"
    size="large"
    color={ color }
    onClick={ onClick }
    disabled={ disabled }
    data-testid={ testId }
  >
    { children }
  </CustomButton>
)