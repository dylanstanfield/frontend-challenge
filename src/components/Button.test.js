import { render } from "@testing-library/react";

import { Button } from "./Button";

describe('Button', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Hello World</Button>);
    expect(container).toMatchSnapshot();
  });
});