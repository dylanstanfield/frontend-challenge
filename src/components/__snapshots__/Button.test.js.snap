// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-3z3v1y-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Hello World
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
